src/cmdc_tools/datasets/official/UT/data.py

```diff
@@ -0,0 +1,175 @@
+            .melt(id_vars=["dt"], var_name="variable_name")
+            .assign(vintage=pd.Timestamp.utcnow(), fips=self.state_fips)
+            .sort_values(["dt", "variable_name"])
+        )
+
+
+class UtahFips(DatasetBaseNoDate):
+    has_fips = True
+    state_fips = int(us.states.lookup("Utah").fips)
+    source = "https://coronavirus-dashboard.utah.gov/#overview"
+
+    def get(self):
+        hosp = self._get_hosp_sync()
+        tests = self._get_tests_sync()
+        deaths = self._get_deaths_sync()
+
+        return pd.concat([hosp, tests, deaths], sort=False)
+
+    async def _get_hosp(self):
+        url = "https://coronavirus-dashboard.utah.gov/#hospitalizations-mortality"
+        async with with_page() as page:
+            await page.goto(url)
+            await page.waitForXPath("//div[@class='plot-container plotly']")
+            plots = await page.Jx(
+                "//div[@id='daily-hospital-survey-previous-8-weeks']//div[@class='plot-container plotly']/.."
+            )
+            text = await page.evaluate("(elem) => [elem.data, elem.layout]", plots[0])
+            data = text[0]
+            layout = text[1]
+            reduced = self._extract_plotly_data(data, layout)
+
+            renamed = reduced.rename(
+                columns={
+                    "ICU": "icu_beds_in_use_any",
+                    "Non-ICU": "hospital_beds_in_use_any",
+                }
+            )
+
+            return (
+                renamed[["dt", "icu_beds_in_use_any", "hospital_beds_in_use_any"]]
+                .melt(id_vars=["dt"], var_name="variable_name")
+                .assign(vintage=pd.Timestamp.utcnow(), fips=self.state_fips)
+            )
+
+    async def _get_tests(self):
+        url = "https://coronavirus-dashboard.utah.gov/#overview"
+
+        async with with_page() as page:
+            await page.goto(url)
+            await page.waitForXPath("//div[@class='plot-container plotly']")
+            plots = await page.Jx(
+                "//div[@id='total-tests-by-date']//div[@class='plot-container plotly']/.."
+            )
+            text = await page.evaluate("(elem) => [elem.data, elem.layout]", plots[0])
+            data = text[0]
+            layout = text[1]
+            # return text
+
+            df = self._extract_plotly_data(data, layout)
+
+            renamed = df.fillna(0)
+
+            renamed["positive_tests_total"] = (
+                renamed["Positive PCR"] + renamed["Positive Antigen"]
+            ).astype(int)
+            renamed["negative_tests_total"] = (
+                renamed["Negative PCR"] + renamed["Negative Antigen"]
+            ).astype(int)
+            sorts = renamed.set_index("dt").sort_index().cumsum().reset_index()
+            return (
+                sorts[["dt", "negative_tests_total", "positive_tests_total"]]
+                .melt(id_vars=["dt"], var_name="variable_name")
+                .assign(vintage=pd.Timestamp.utcnow(), fips=self.state_fips)
```

Comment:
We have a new method for fetching the vintage so that we are consistent in how we set the vintage. See `DatasetBaseNoDate._retrieve_vintage`